src/cliffordsynthesis/TargetMetricHandler.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+* This file is part of the MQT QMAP library which is released under the MIT license.
+* See file README.md or go to https://www.cda.cit.tum.de/research/ibm_qx_mapping/ for more information.
+*/
+
+#include "cliffordsynthesis/TargetMetricHandler.hpp"
+
+#include "cliffordsynthesis/GateEncoding.hpp"
+namespace cs {
+
+    void TargetMetricHandler::makeTargetMetric(const SynthesisData& data, const Configuration& configuration) {
+        bool useMaxSat = configuration.strategy == OptimizationStrategy::UseMinimizer || configuration.strategy == OptimizationStrategy::SplitIter;
+        bool onlyCnot  = configuration.target == TargetMetric::TWO_QUBIT_GATES;
+
+        switch (configuration.target) {
+            case TargetMetric::GATES:
+            case TargetMetric::TWO_QUBIT_GATES:
+                makeGateMetric(data, useMaxSat, onlyCnot);
+                break;
+            case TargetMetric::DEPTH:
+                makeDepthMetric(data, useMaxSat);
+                break;
+            case TargetMetric::FIDELITY:
+                makeFidelityMetric(data, useMaxSat, configuration.architecture);
+                break;
+        }
+    }
+    void TargetMetricHandler::makeGateMetric(const SynthesisData& data, bool useMaxSat, bool onlyCNOT) {
+        logicbase::LogicTerm changes = logicbase::LogicTerm(true);
+        // COST
+        if (useMaxSat) {
+            logicbase::LogicTerm cost = logicbase::LogicTerm(0);
+            for (unsigned int gateStep = 1; gateStep < data.timesteps + 1; ++gateStep) {
+                for (unsigned int a = 0; a < data.nqubits; ++a) {
+                    if (!onlyCNOT) {
+                        for (auto gate: Gates::SINGLE_QUBIT_WITHOUT_NOP) {
+                            cost = cost + data.gS[gateStep][Gates::toIndex(gate)][a];
+                        }
+                    }
+                    for (unsigned int b = 0; b <= a; ++b) {
```

Comment:
use `std::size_t` instead of `unsigned int`. Please check throughout this file for similar occurrences